dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/db/sql/DorisSqlBuilder.java

```diff
@@ -354,8 +388,82 @@ public String dropCatalogIfExists() {
     return String.format("drop catalog if exists %s;", quote(catalog));
   }
 
+  @Override
+  public SqlFunction getDateDiffInDays(String date1, String date2) {
+    return () -> "DATEDIFF(" + date1 + ", " + date2 + ")";
+  }
+
+  @Override
+  public String fixQuote(String column) {
+    // Handle null or empty cases
+    if (column == null || column.trim().isEmpty()) {
+      return "";
+    }
+
+    // Trim the entire input first
+    column = column.trim();
+
+    // If it contains a function call (contains '(' and ')'), return as is
+    if (column.contains("(") && column.contains(")")) {
+      return column;
+    }
+
```

Comment:
I am just curious, could we explicitly manage quoting inside the code generating classes, instead of matching on patterns after it is generated? I know it is probably intrusive to include `SqlBuilder`. Though this approach may be a bit coarse and error-prone? E.g. in DHIS 2, for tracker/event analytics, we can search and match attribute values, and what if the an attribute value, like an identifier, contains a `.` ? Or an " as " ?